docs/conf.py

```diff
@@ -68,6 +68,7 @@
     ("py:class", "'id'"),
     ("py:class", "typing_extensions.Literal"),
     ("py:func", "int"),
+    ("py:class", "asyncio.locks.Event"),
```

Comment:
Including this exception breaks the Docs link to this object.

It's certainly weird why it's happening. `asyncio.Event` is a valid target, according to the `objects.inv` file provided by the Python documentation (run `python -m sphinx.ext.intersphinx https://docs.python.org/3/objects.inv > objects.temp`) and inspect the list of valid targets).

I don't understand why Sphinx is insisting that it links to the `.locks` subfolder, when `Event` is [clearly documented](https://docs.python.org/3/library/asyncio-sync.html#asyncio.Event). 

Ultimately it would be good to get this cross-reference working in some way. Perhaps working out how to specify a link in Sphinx from the conf.py file, or similar.

Reply:
What's even weirder is that the `asyncio.Event` reference inside the text description of `flush_event` DOES link to the Python documentation!

Reply:
Yea I was confused by this for a long time (and still am).

`asyncio` should have `Event` in it's `__all__`. Event if it's trying to get it from `asyncio.locks` it should still be working...

``` python
# This relies on each of the submodules having an __all__ variable.
from .base_events import *
from .coroutines import *
from .events import *
from .exceptions import *
from .futures import *
from .locks import *
from .protocols import *
from .runners import *
from .queues import *
from .streams import *
from .subprocess import *
from .tasks import *
from .threads import *
from .transports import *

__all__ = (base_events.__all__ +
           coroutines.__all__ +
           events.__all__ +
           exceptions.__all__ +
           futures.__all__ +
           locks.__all__ +
           protocols.__all__ +
           runners.__all__ +
           queues.__all__ +
           streams.__all__ +
           subprocess.__all__ +
           tasks.__all__ +
           threads.__all__ +
           transports.__all__)
```

Reply:
Might it be helpful to try linking against other `asyncio` types that are actually defines in submodules, e.g. `queues` or `events`?

Reply:
I think this is autodoc trying to be clever and failing. It finds an `Event` in the type signature, finds that its source is `asyncio,locks.Event` and creates a link to that. Intersphinx then looks in the python sphinx docs, and says "I see an `asyncio.Event` but no `asyncio.locks.Event`" and prints the warning. There seems to be no way to tell autodoc where this thing actually is, it is a decision of the python docs to publish under `asyncio.Event` rather than `asyncio.locks.Event` rather than anything it can get from the code.

This is a docs problem that has taken more than 30 minutes with no fix, so I will invoke my "too hard, put in an ignore and move on" rule...

Reply:
Agreed. It still links the `asyncio.Event` with the nitpick ignore so I say we should just leave it in.